backend/lib/services/shoots.js

```diff
@@ -211,61 +156,60 @@ exports.replaceMaintenance = async function ({ user, namespace, name, body }) {
       }
     }
   }
-  return patch({ user, namespace, name, body: payload })
+  return client['core.gardener.cloud'].shoots.mergePatch(namespace, name, payload)
 }
 
 const patchAnnotations = async function ({ user, namespace, name, annotations }) {
+  const client = user.client // user specific client for the garden cluster
   const body = {
     metadata: {
       annotations: annotations
     }
   }
-  return patch({ user, namespace, name, body })
+  return client['core.gardener.cloud'].shoots.mergePatch(namespace, name, body)
 }
 exports.patchAnnotations = patchAnnotations
 
 exports.remove = async function ({ user, namespace, name }) {
+  const client = user.client // user specific client for the garden cluster
   const annotations = {
     'confirmation.garden.sapcloud.io/deletion': 'true'
   }
   await patchAnnotations({ user, namespace, name, annotations })
 
-  return Garden(user).namespaces(namespace).shoots.delete({ name })
+  return client['core.gardener.cloud'].shoots.delete(namespace, name)
+}
+
+async function readSecret ({ user, namespace, name }) {
+  const client = user.client // user specific client for the garden cluster
+  try {
+    return await client.core.secrets.get(namespace, name)
```

Comment:
catch http errors and return null in client

Reply:
added to client but not to mixins.
✔️